main/channel.c

```diff
@@ -1383,14 +1456,33 @@ struct ast_channel_iterator *ast_channel_iterator_by_name_new(const char *name,
 {
 	struct ast_channel_iterator *i;
 	char *l_name = (char *) name;
+	struct ast_channel_callback_data callback_data = {
+		.search_length		= &name_len,
+		.final_match_result	= 0
+	};
 
 	if (!(i = ast_calloc(1, sizeof(*i)))) {
 		return NULL;
 	}
 
-	i->active_iterator = (void *) ast_channel_callback(ast_channel_by_name_cb,
-		l_name, &name_len,
-		OBJ_MULTIPLE | (name_len == 0 /* match the whole word, so optimize */ ? OBJ_KEY : 0));
+	for (;;) {
```

Comment:
This logic seems duplicated several times, could it be refactored?

Reply:
I'll see what I can do!